modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -41,7 +43,20 @@ def run_plan(scope, plan_name, named_args = {})
         old_run_as = executor.run_as
         executor.run_as = run_as
       end
-      result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, use_args, true)
+      if named_args['_catch_errors']
+        begin
+          result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, use_args, true)
+        rescue Puppet::PreformattedError => err
+          # CODEREVIEW: should this be ducktyped?
```

Comment:
I'm not sure why that would be helpful. It would have to implement `to_puppet_error`? I'd at least want a more sensible interface if it's something we expect people to extend.